pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,719 @@
+	}
+
+	return x.submitTransactionNotary(simpleAcc, []actor.SignerAccount{accountToSignerAccount(multiSigAcc)}, func(*notary.Actor) (*transaction.Transaction, error) {
+		return &tx, nil
+	})
+}
+
+// CallContractNotarized creates transaction for calling specified contract
+// (particular method with its arguments), wraps the transaction into request
+// for signing by:
+//  1. Proxy contract deployed in the NeoFS Sidechain
+//  2. a group of parties presented in a provided keys.PublicKeys
+//  3. Notary native contract
+//
+// and spreads it in the blockchain network. The action is performed via Neo
+// Notary request. The request is signed on behalf of the specified simple Neo
+// account.
+//
+// Passed keys.PublicKeys must not be empty.
+func (x *Blockchain) CallContractNotarized(simpleAcc *wallet.Account, signerKeys keys.PublicKeys, proxyContract, calledContract util.Uint160, method string, args ...interface{}) error {
+	if len(signerKeys) == 0 {
+		panic("missing signer keys")
+	}
+
+	multiSigAcc, err := formMultiSigAccountFromKeys(simpleAcc, signerKeys)
+	if err != nil {
+		return err
+	}
+
+	signers := []actor.SignerAccount{
+		contractSignerAccount(proxyContract),
+		accountToSignerAccount(multiSigAcc),
+		contractSignerAccount(notary.Hash),
+	}
+
+	return x.submitTransactionNotary(simpleAcc, signers, func(a *notary.Actor) (*transaction.Transaction, error) {
+		return a.MakeCall(calledContract, method, args...)
+	})
+}
+
+// submitTransactionNotary constructs transaction.Transaction and wraps it into
+// request for signing by a group of parties and spreads the request in the
+// blockchain network. The action is performed via Neo Notary service. The
+// request is signed on behalf of the specified simple Neo account.
+func (x *Blockchain) submitTransactionNotary(simpleAcc *wallet.Account, signers []actor.SignerAccount, fTx func(*notary.Actor) (*transaction.Transaction, error)) error {
+	a, err := notary.NewActor(x.rpcActor, signers, simpleAcc)
+	if err != nil {
+		return fmt.Errorf("init notary actor: %w", err)
+	}
+
+	_, _, _, err = a.Notarize(fTx(a))
+	if err != nil {
+		return fmt.Errorf("notarize transaction using notary actor: %w", err)
+	}
+
+	return nil
+}
+
+// Height returns sequence number of the latest block in the blockchain.
+func (x *Blockchain) Height() uint32 {
+	return x.base.BlockHeight()
+}
+
+// SubscribeToNotifications performs subscription to notification events spawned
+// by Neo smart contracts and stream them into the returned channel as
+// state.ContainedNotificationEvent instances. Caller must read the channel
+// regularly in the background to prevent congestion. When notifications are no
+// longer needed, caller should cancel subscription by calling the returned
+// function.
+//
+// SubscribeToNotifications is expected to be called once since event
+// multiplexing does not make sense.
+func (x *Blockchain) SubscribeToNotifications() (<-chan *state.ContainedNotificationEvent, func()) {
```

Comment:
No longer needed, regular subscriptions via Internal are the same as via WSClient.